internal/mode/static/nginx/file/manager_test.go

```diff
@@ -116,6 +116,27 @@ var _ = Describe("EventHandler", func() {
 		})
 	})
 
+	When("file does not exist", func() {
+		It("should not error", func() {
+			fakeOSMgr := &filefakes.FakeOSFileManager{}
+			tmpDir := GinkgoT().TempDir()
+			mgr := file.NewManagerImpl(zap.New(), fakeOSMgr)
+
+			files := []file.File{
+				{
+					Type:    file.TypeRegular,
+					Path:    filepath.Join(tmpDir, "regular-1.conf"),
+					Content: []byte("regular-1"),
+				},
+			}
+
+			Expect(mgr.ReplaceFiles(files)).ToNot(HaveOccurred())
+			fakeOSMgr.RemoveReturns(os.ErrNotExist)
```

Comment:
nit: because `fakeOSMgr.RemoveReturns(os.ErrNotExist)`  is more relevant to line 136 (second call to ReplaceFiles) rather than 133, I'd move it closer to 136 :  this will help the reader interpret what we're doing it with less efforts:

```go
			Expect(mgr.ReplaceFiles(files)).ToNot(HaveOccurred())
			
			fakeOSMgr.RemoveReturns(os.ErrNotExist)
			Expect(mgr.ReplaceFiles(files)).ToNot(HaveOccurred())
```